src/openms/source/KERNEL/MSExperiment.cpp

```diff
@@ -777,6 +818,19 @@ namespace OpenMS
     return false;
   }
 
+  bool MSExperiment::isIMFrame() const
+  {
+    if (spectra_.empty()) return false;
+    auto rt_start = spectra_[0].getRT();
+    auto last_drift = std::numeric_limits<double>::lowest();
+    for (const auto& s : spectra_) {
+      if (s.getRT() != rt_start) return false; // RT changes...
+      if (s.getDriftTime() == last_drift) return false; // IM did not change...
```

Comment:
this disallows equal IM values in IMFrames. Is it not possible to have two peaks with the same IM by collapsing a range of m/z values? or is it assumed here, that some kind of filtering for equal values has been performed for those cases?